### ISSUE TITLE
Possible input analysis incomplete

### ISSUE THREAD

**Issue by matthewpersico** [koalaman/shellcheck#2441]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2086
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

declare a,b

if (( $# ==2 )); then
        a=$1
        b=$2
else
        a=constant
        b=$1
fi

somefunc $a
somefunc $b

```

#### Here's what shellcheck currently says:
```
Line 14	SC2086: Double quote to prevent globbing and word splitting.
```
#### Here's what I wanted or expected to see:

```
Line 13	SC2086: Double quote to prevent globbing and word splitting.
Line 14	SC2086: Double quote to prevent globbing and word splitting.
```

It appears that if the latest assignment is a constant, the check doesn't "remember" that there was a previous non-constant one. Contrast this with the inverse:

```sh
#!/bin/bash

declare a,b

if (( $# ==2 )); then
        a=constant
        b=$2
else
        a=$2
        b=$1
fi

somefunc $a
somefunc $b

```

#### Here's what shellcheck currently says:
```
Line 13	SC2086: Double quote to prevent globbing and word splitting.
Line 14	SC2086: Double quote to prevent globbing and word splitting.
```

**Comment by koalaman** [comment]
Finally fixed in #2541!
